modules/drivers/bigquery/src/metabase/driver/bigquery/query_processor.clj

```diff
@@ -436,6 +436,10 @@
          (>= (count components) 2))
     true))
 
+(defmethod sql.qp/cast-temporal-string [:bigquery :Coercion/YYYYMMDDHHMMSSString->Temporal]
+  [_driver _coercion_strategy expr]
```

Comment:
What's the reason behind using `_snake_case` instead of just `_` or even `_coercion-strategy`? It's really making me itchy

Reply:
it comes off a field so it's properly `coercion_strategy` there, it's unused so prepend `_`. I'm a big fan of named underscores. I've spent time wondering how to get at some information and only later realized that it was trivial based on one of the unused parameters so i prefer `_name` to just `_` to remember what things are. But I don't have a strong opinion here. Happy to switch to `_` or `_coercion-strategy` as you like.

Reply:
I like `_coercion-strategy` a lot better. Our general pattern is to only use `snake_case` variable names when they come directly from API endpoints or the app DB. Once they get passed to another function there's no reason to keep using `snake_case`.

On a side note, one of the things I have working in Blue JDBC are the "magic maps" I experimented with a while back that are case and `lisp-case`/`snake_case` insensitive. So eventually we'll be able to use `lisp-case` everywhere in the Clojure code, and it will do the right thing automatically when hitting the DB or API boundaries. With that in mind, we shouldn't propogate the snake_case variables around the code base too much

Reply:
Done. I think i'm a bit hesitant on the magic maps. I'm a bit wary of things that try to be a bit too clever. I'd prefer to be more deliberate on the seams where things would have snake case vs lisp case and why and clean all those up. But we'll see how i like it in practice.

Reply:
I've cleaned up all the argument position `coercion_strategy` -> `coercion-strategy`, including ones with underscores prepended